### PR TITLE
Fix negative number support

### PR DESCRIPTION
**Pull request by Jeff-Ciesielski**
@keyme/robotics-engineers

### CODE REVIEW

File: src/bitvec.nim [hunk 1]
```diff
@@ -25,11 +25,11 @@ proc encode*[T: SomeInteger](n: T): seq[byte] =
   if n == 0:
     return @[0.byte]
 
-  var num = n
-  while num > 0.T:
+  var num = cast[uint64](n)
```
Comment (shonfeder) [inline src/bitvec.nim]: Could you leave a comment indicating why a cast (as opposed to a conversion) is needed here?

**Review by shonfeder — COMMENTED** [review]
LGTM otherwise

**Comment by Jeff-Ciesielski** [comment]
Comment addressed

**Review by shonfeder — APPROVED** [review]
LGTM

**Pull request merged**